tools/src/main/scala/com/nvidia/spark/rapids/tool/EventLogPathProcessor.scala

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.tool
+
+import java.io.FileNotFoundException
+import java.time.LocalDateTime
+import java.util.zip.ZipOutputStream
+
+import scala.collection.mutable.LinkedHashMap
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FileStatus, FileSystem, Path}
+
+import org.apache.spark.deploy.history.{EventLogFileReader, EventLogFileWriter}
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.rapids.tool.profiling.ApplicationInfo
+
+sealed trait EventLogInfo {
+  def eventLog: Path
+}
+
+case class ApacheSparkEventLog(override val eventLog: Path) extends EventLogInfo
+case class DatabricksEventLog(override val eventLog: Path) extends EventLogInfo
+
+
+object EventLogPathProcessor extends Logging {
+  // Apache Spark event log prefixes
+  val EVENT_LOG_DIR_NAME_PREFIX = "eventlog_v2_"
+  val EVENT_LOG_FILE_NAME_PREFIX = "events_"
+
+  def isEventLogDir(status: FileStatus): Boolean = {
+    status.isDirectory && status.getPath.getName.startsWith(EVENT_LOG_DIR_NAME_PREFIX)
+  }
+
+  // This only checks the name of the path
+  def isEventLogDir(path: String): Boolean = {
+    path.startsWith(EVENT_LOG_DIR_NAME_PREFIX)
+  }
+
+  // https://github.com/apache/spark/blob/0494dc90af48ce7da0625485a4dc6917a244d580/
+  // core/src/main/scala/org/apache/spark/io/CompressionCodec.scala#L67
+  val SPARK_SHORT_COMPRESSION_CODEC_NAMES = Set("lz4", "lzf", "snappy", "zstd")
+  // Apache Spark ones plus gzip
+  val SPARK_SHORT_COMPRESSION_CODEC_NAMES_FOR_FILTER =
+    SPARK_SHORT_COMPRESSION_CODEC_NAMES ++ Set("gz")
+
+  def eventLogNameFilter(logFile: Path): Boolean = {
+    EventLogFileWriter.codecName(logFile)
+      .forall(suffix => SPARK_SHORT_COMPRESSION_CODEC_NAMES_FOR_FILTER.contains(suffix))
+  }
+
+  // Databricks has the latest events in file named eventlog and then any rolled in format
+  // eventlog-2021-06-14--20-00.gz
+  def isDatabricksEventLogDir(dir: FileStatus, fs: FileSystem): Boolean = {
+    // try to determine if dir structure looks right
+    val dirList = fs.listStatus(dir.getPath)
+    if (dirList.size > 0) {
+      if (dirList.exists(_.getPath.getName.equals("eventlog"))) {
+        if (dirList.size > 1) {
+          dirList.exists(_.getPath.getName
+            .matches("eventlog-([0-9]){4}-([0-9]){2}-([0-9]){2}--([0-9]){2}-([0-9]){2}.*"))
```

Comment:
this will keep recompiling regex. We need to create a pattern object outside the dir list loop. e.g. via `"regex".r`  or java `Pattern.compile`. 

we could also pass a `PathFilter` instance to `listStatus`  accepting:  `name  == eventlog || matches(regex)` . Then the return value of the `isDatabricksEventLogDir` is `res.contains("eventlog") && res.size() > 1`

Reply:
yeah we probably don't even need the regex based on how everything else is done now, I'll update